components/serviceMap/customNodes.tsx

```diff
@@ -0,0 +1,441 @@
+import { Handle, MarkerType, Position } from "reactflow";
+import IconGripVertical from "tabler-icons/tsx/grip-vertical.tsx";
+import IconDatabase from "tabler-icons/tsx/database.tsx";
+import "flowbite";
+import "twind";
+import { useState } from "preact/hooks";
+import { Audience, OperationType } from "snitch-protos/protos/common.ts";
+import { NodeMenu, ServiceNodeMenu } from "./nodeMenu.tsx";
+import { NodeData } from "../../islands/serviceMap.tsx";
+import { ProducerIcon } from "../icons/producer.tsx";
+import { ConsumerIcon } from "../icons/consumer.tsx";
+import { titleCase } from "../../lib/utils.ts";
+import { Pipeline } from "snitch-protos/protos/pipeline.ts";
+import { PipelineInfo } from "snitch-protos/protos/info.ts";
+
+export type AudiencePipeline = Audience & { pipeline?: Pipeline };
```

Comment:
Adding pipeline information to the audience and adding that to the nodes so all flow nodes will have access to that.